mech_client/marketplace_interact.py

```diff
@@ -220,11 +238,13 @@ def fetch_requester_nvm_subscription_balance(
     :type ledger_api: EthereumApi
     :param mech_payment_balance_tracker: Requested mech's balance tracker contract address
     :type mech_payment_balance_tracker: str
+    :param payment_type: Requested mech's payment type
+    :type payment_type: str
     :return: The requester balance.
     :rtype: int
     """
     with open(
-        Path(__file__).parent / "abis" / "BalanceTrackerNvmSubscriptionNative.json",
+        PAYMENT_TYPE_TO_ABI_PATH[payment_type],
```

Comment:
Access the dict safely.

Reply:
Should be safe as I am checking for payment types here: https://github.com/valory-xyz/mech-client/blob/feat/nvm-token/mech_client/marketplace_interact.py#L780

Reply:
You mean here right?

https://github.com/valory-xyz/mech-client/blob/eb3f136b76f9599249a380226c432a3d8b87690b/mech_client/marketplace_interact.py#L775

Yeap, makes sense.